src/Agent.Listener/Configuration/InteractiveSessionConfigurationManager.cs

```diff
@@ -0,0 +1,213 @@
+            //shutdown reason
+            var shutdownReasonValue = regHelper.GetRegistry(WellKnownRegistries.ShutdownReason);
+            int.TryParse(shutdownReasonValue, out int shutdownReasonOn);
+            if(shutdownReasonOn == 1)
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_ShutdownReason"));
+            }
+
+            var legalNoticeCaption = regHelper.GetRegistry(WellKnownRegistries.LegalNoticeCaption);
+            var legalNoticeText =  regHelper.GetRegistry(WellKnownRegistries.LegalNoticeText);
+            if(!string.IsNullOrEmpty(legalNoticeCaption) || !string.IsNullOrEmpty(legalNoticeText))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_LegalNotice"));
+            }
+
+            //auto-logon
+            var autoLogonCountValue = regHelper.GetRegistry(WellKnownRegistries.AutoLogonCount);            
+            if(!string.IsNullOrEmpty(autoLogonCountValue))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_AutoLogonCount"));
+            }
+
+            if(warningReasons.Count > 0)
+            {
+                _terminal.WriteLine(StringUtil.Loc("UITestingWarning"));
+                for(int i=0; i < warningReasons.Count; i++)
+                {
+                    _terminal.WriteLine(String.Format("{0} - {1}", (i+1).ToString(), warningReasons[i]));
+                }
+            }
+        }
+
+        private void UpdateRegistrySettingsforUITesting(WindowsRegistryHelper regHelper, string userName, string domainName, string logonPassword)
+        {
+            regHelper.SetRegistry(WellKnownRegistries.ScreenSaver, "0");
+            regHelper.SetRegistry(WellKnownRegistries.ScreenSaverDomainPolicy, "0");
+
+            ShowAutoLogonWarningIfAlreadyEnabled(regHelper, userName);
+
+            var windowsHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            windowsHelper.SetAutoLogonPassword(logonPassword);
+
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogonUserName, userName);
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogonDomainName, domainName);
+            regHelper.SetRegistry(WellKnownRegistries.AutoLogon, "1");
+
+            var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agent.service.exe");
+            var startupCommand = string.Format("{0} {1}", startupProcessPath, "runAsProcess");
+            regHelper.SetRegistry(WellKnownRegistries.StartupProcess, startupCommand);
+
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReason, "0");
+            regHelper.SetRegistry(WellKnownRegistries.ShutdownReasonUI, "0");
+
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeCaption, "");
+            regHelper.SetRegistry(WellKnownRegistries.LegalNoticeText, "");
+
+            var processInvoker = HostContext.GetService<IProcessInvoker>();
+            processInvoker.ExecuteAsync(
```

Comment:
not: registry settings in one method and configuration of powerconfig in another method.